entities/magnetic_susceptibility/EnsembleAverager.h

```diff
@@ -0,0 +1,22 @@
+#ifndef JULY_ENSEMBLEAVERAGER_H
+#define JULY_ENSEMBLEAVERAGER_H
+
+#include <cmath>
+
+#include "entities/data_structures/DenseMatrix.h"
+namespace magnetic_susceptibility {
+class EnsembleAverager {
+  public:
+    EnsembleAverager(DenseVector&& energy, DenseVector&& degeneracy);
+    double ensemble_average(const DenseVector& value, double temperature) const;
+
+  private:
+    const DenseVector energy_;
+    const DenseVector degeneracy_;
+    mutable DenseVector divided_and_wise_exped_energy;
```

Comment:
why mutable?

Reply:
These three fields are helping to avoid recalculating the same values. If temperature in ensemble_average() function is the same as in the previous step, they should stay the same and vice versa.